app/src/main/java/com/group4sweng/scranplan/MealPlanner/PlannerFragment.java

```diff
@@ -146,20 +157,34 @@ private void openRecipeDialog(SearchQuery query) {
         plannerListFragment.show(getParentFragmentManager(), "search");
     }
 
-    //Sets default paramters for buttons
-    private void defaultButton(final ImageButton imageButton) {
+    //Sets meal timescale parameters for buttons
+    private void mealTimescaleButton(final ImageButton imageButton, int i) {
         imageButton.setImageResource(R.drawable.add); //Default image
         imageButton.setOnClickListener(v -> {
             Bundle bundle = new Bundle();
             bundle.putBoolean("planner", true); //Condition to let child fragments know access is from planner
             currentSelection = imageButton; //Allows tracking of button pressed
 
             //Creates and launches recipe fragment
-            recipeFragment = new RecipeFragment(mUser);
-            recipeFragment.setArguments(bundle);
-            recipeFragment.setTargetFragment(PlannerFragment.this, 1);
-            fragmentTransaction = getParentFragmentManager().beginTransaction();
-            fragmentTransaction.add(R.id.frameLayout, recipeFragment); //Overlays fragment on existing one
+            if (i == 0) {
+                breakfastFragment = new BreakfastFragment(mUser);
+                breakfastFragment.setArguments(bundle);
+                breakfastFragment.setTargetFragment(PlannerFragment.this, breakfastRequestCode);
+                fragmentTransaction = getParentFragmentManager().beginTransaction();
+                fragmentTransaction.add(R.id.frameLayout, breakfastFragment); //Overlays fragment on existing one
+            }else if (i == 1){
+                lunchFragment = new LunchFragment(mUser);
+                lunchFragment.setArguments(bundle);
+                lunchFragment.setTargetFragment(PlannerFragment.this, lunchRequestCode);
+                fragmentTransaction = getParentFragmentManager().beginTransaction();
+                fragmentTransaction.add(R.id.frameLayout, lunchFragment); //Overlays fragment on existing one
+            }else if (i == 2){
+                dinnerFragment = new DinnerFragment(mUser);
+                dinnerFragment.setArguments(bundle);
+                dinnerFragment.setTargetFragment(PlannerFragment.this, dinnerRequestCode);
+                fragmentTransaction = getParentFragmentManager().beginTransaction();
+                fragmentTransaction.add(R.id.frameLayout, dinnerFragment); //Overlays fragment on existing one
+            }
```

Comment:
Still really confused why we are using 3 fragments for basically the same thing with different text + images.
It does work though I guess.